drivers/watchdog/wdt_dw.c

```diff
@@ -11,55 +11,77 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/sys_clock.h>
 #include <zephyr/math/ilog2.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/clock_control.h>
 
 #include "wdt_dw.h"
 #include "wdt_dw_common.h"
 
 LOG_MODULE_REGISTER(wdt_dw, CONFIG_WDT_LOG_LEVEL);
 
-#define WDT_IS_INST_IRQ_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), interrupts)
-#define WDT_CHECK_INTERRUPT_USED(inst)	WDT_IS_INST_IRQ_EN(inst) ||
-#define WDT_DW_INTERRUPT_SUPPORT	DT_INST_FOREACH_STATUS_OKAY(WDT_CHECK_INTERRUPT_USED) 0
-
 /* Device run time data */
 struct dw_wdt_dev_data {
+	/* MMIO mapping information for watchdog register base address */
+	DEVICE_MMIO_RAM;
+	/* Clock frequency */
+	uint32_t clk_freq;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(clocks)
+	/* Clock controller dev instance */
+	const struct device *clk_dev;
+	/* Identifier for timer to get clock freq from clk manager */
+	clock_control_subsys_t clkid;
+#endif
```

Comment:
The clock controller information (if existing) is still constant and can stay in the config struct, right?

Reply:
yeah, clock controller information is constant always. Just to make it look simple, I created it in the data struct. I can make these changes and add it in config struct